### PR TITLE
Mention how to target unregistered biomes

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
Was suggested in here to mention how to handle Pale Garden biome
https://github.com/neoforged/NeoForge/issues/1649#issuecomment-2452598540

------------------
Preview URL: https://pr-189.neoforged-docs-previews.pages.dev

### CODE REVIEW

File: docs/worldgen/biomemodifier.md
Comment (IchHabeHunger54) [inline docs/worldgen/biomemodifier.md:716]:
```suggestion
Using that biome tag for a biome modifier will now not crash if the biome is not registered. A use case for this is the Pale Garden biome in 1.21.3. That biome is only created when you turn on the Winter Drop datapack in-game. Otherwise, the biome does not exist in the biome registry at all. Another use case can be to target modded biomes while still functioning when the mods adding these biomes are not present.
```

**Review by IchHabeHunger54 — CHANGES_REQUESTED** [review]
Looks good, one small nitpick.

**Review by IchHabeHunger54 — DISMISSED** [review]
Approved, but merging only once #178 has been merged as well.

File: docs/worldgen/biomemodifier.md
Comment (ChampionAsh5357) [inline docs/worldgen/biomemodifier.md:714]: I would also provide a method for how to datagen this using a `TagsProvider`

Comment (TelepathicGrunt) [reply docs/worldgen/biomemodifier.md:714]: I don’t know how to datagen. Would you be able to add a commit to this PR of what that should look like?

Comment (ChampionAsh5357) [reply docs/worldgen/biomemodifier.md:714]:
```java
// In a TagsProvider<Biome> subclass
// Assume we have some example TagKey<Biome> OPTIONAL_BIOMES_TAG
@Override
protected void addTags(HolderLookup.Provider registries) {
    this.tag(OPTIONAL_BIOMES_TAG)
        // Must be a ResourceLocation representing the registry object
        .addOptional(WinterDropBiomes.PALE_GARDEN.location());
}
```

Should be something like that.

**Comment by IchHabeHunger54** [comment]
Rebase, then this should be good to go.

**Review by IchHabeHunger54 — APPROVED** [review]

**Review by IchHabeHunger54 — APPROVED** [review]

**Pull request merged**
